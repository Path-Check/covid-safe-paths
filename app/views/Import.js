import React, { Component } from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  Linking,
  View,
  Text,
  Alert
} from 'react-native';

import colors from "../constants/colors";
import { WebView } from 'react-native-webview';
import Button from "../components/Button";
import BackgroundGeolocation from '@mauron85/react-native-background-geolocation';

class ImportScreen extends Component {
    constructor(props) {
        super(props);
    }
    componentDidMount() {

        /*BackgroundGeolocation.on('location', (location) => {


            GetStoreData('LOCATION_DATA')
            .then(locationArray => {
                var locationData;

                if (locationArray !== null) {
                  locationData = JSON.parse(locationArray);
                } else {
                  locationData = [];
                }

                locationData.push(location);
                SetStoreData('LOCATION_DATA', locationData);
            });

            // to perform long running operation on iOS
            // you need to create background task
            BackgroundGeolocation.startTask(taskKey => {
                // execute long running task
                // eg. ajax post location
                // IMPORTANT: task has to be ended by endTask
                BackgroundGeolocation.endTask(taskKey);
            });
        });

        BackgroundGeolocation.on('stationary', (stationaryLocation) => {
            // handle stationary locations here
            // Actions.sendLocation(stationaryLocation);
            console.log('[INFO] stationaryLocation:', stationaryLocation);
        });

        BackgroundGeolocation.on('error', (error) => {
        console.log('[ERROR] BackgroundGeolocation error:', error);
        });

        BackgroundGeolocation.on('start', () => {
        console.log('[INFO] BackgroundGeolocation service has been started');
        });

        BackgroundGeolocation.on('stop', () => {
        console.log('[INFO] BackgroundGeolocation service has been stopped');
        });

        BackgroundGeolocation.on('authorization', (status) => {
        console.log('[INFO] BackgroundGeolocation authorization status: ' + status);
        if (status !== BackgroundGeolocation.AUTHORIZED) {
            // we need to set delay or otherwise alert may not be shown
            setTimeout(() =>
            Alert.alert('App requires location tracking permission', 'Would you like to open app settings?', [
                { text: 'Yes', onPress: () => BackgroundGeolocation.showAppSettings() },
                { text: 'No', onPress: () => console.log('No Pressed'), style: 'cancel' }
            ]), 1000);
        }
        });

        BackgroundGeolocation.on('background', () => {
        console.log('[INFO] App is in background');
        });

        BackgroundGeolocation.on('foreground', () => {
        console.log('[INFO] App is in foreground');
        });

        BackgroundGeolocation.on('abort_requested', () => {
        console.log('[INFO] Server responded with 285 Updates Not Required');

        // Here we can decide whether we want stop the updates or not.
        // If you've configured the server to return 285, then it means the server does not require further update.
        // So the normal thing to do here would be to `BackgroundGeolocation.stop()`.
        // But you might be counting on it to receive location updates in the UI, so you could just reconfigure and set `url` to null.
        });

        BackgroundGeolocation.on('http_authorization', () => {
        console.log('[INFO] App needs to authorize the http requests');
        });

        // you can also just start without checking for status
        // BackgroundGeolocation.start();*/
    }

    componentWillUnmount() {
        // unregister all event listeners
        BackgroundGeolocation.removeAllListeners();
    }

    render() {
        return (
            <>
            <View style={styles.main}>
                <View style={styles.headerTitle}>
                        <Text style={styles.sectionDescription, {fontSize: 22, marginTop: 8}}>Import Data:</Text>
                </View>
                <View style={styles.subHeaderTitle}>
                        <Text style={styles.sectionDescription, {fontSize: 18, marginTop: 8}}>Rolling out soon</Text>
                </View>
                <View style={styles.web}>
                    <WebView
                        source= {{ uri: 'http://privatekit.mit.edu' }}
                        style= {{ marginTop: 15, marginLeft: 15}}
                    />
                </View>
            </View>
            <View style={styles.footer}>
                <Text style={styles.sectionDescription, { textAlign: 'center', paddingTop: 15 }}>For more information visit the Private Kit hompage:</Text>
                <Text style={styles.sectionDescription, { color: 'blue', textAlign: 'center' }} onPress={() => Linking.openURL('https://privatekit.mit.edu')}>privatekit.mit.edu</Text>
            </View>
        </>
        )
    }
}

const styles = StyleSheet.create({
    // Container covers the entire screen
    container: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        color: colors.PRIMARY_TEXT,
        backgroundColor: colors.APP_BACKGROUND,
    },
    headerTitle: {
        textAlign: 'center',
        fontWeight: "bold",
        fontSize: 38,
        padding: 0
    },
    subHeaderTitle: {
        textAlign: 'center',
        fontWeight: "bold",
        fontSize: 22,
        padding: 5
    },
    web: {
        flex: 1,
        width: "95%"
    },
    main: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        width: "95%"
    },
    block: {
      margin: 20,
      width: "100%"
    },
    topView: {
        flex: 1,
    },
    footer: {
        textAlign: 'center',
        fontSize: 12,
        fontWeight: '600',
        padding: 4,
        paddingBottom: 10
    },
    intro: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'stretch',
    },
    sectionDescription: {
      fontSize: 18,
      lineHeight: 24,
      fontWeight: '400',
      marginTop: 20,
      marginLeft: 10,
      marginRight: 10
    }
  });
export default ImportScreen;
