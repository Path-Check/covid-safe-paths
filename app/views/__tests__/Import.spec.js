import { fireEvent, render, NativeTestEvent, act } from '@testing-library/react-native'
import 'react-native';
import {Linking} from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Import from '../Import';

jest.spyOn(console, 'log').mockImplementation(() => {});

const GoogleTakeOutAutoImport = require('../../helpers/GoogleTakeOutAutoImport');
const General = require('../../helpers/General');

const navigationMock = {
  goBack: jest.fn()
};
const openURLSpy = jest.spyOn(Linking, 'openURL').mockImplementation(() => Promise.resolve());
const importTakeoutDataSpy = jest.spyOn(GoogleTakeOutAutoImport, 'importTakeoutData')
  .mockImplementation(() => Promise.resolve([]));
const pickFileSpy = jest.spyOn(General, 'pickFile')
  .mockImplementation(() => Promise.resolve('file://tmp/test.zip'));

describe('Import component', () => {
  afterAll(() => {
    openURLSpy.mockReset();
    openURLSpy.mockRestore();
    jest.resetModules();
  });

  it('renders correctly', () => {
    const tree = renderer
    .create(<Import navigation={navigationMock} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('clicking on Google Takeout button leads to open browser page', () => {
    const { getByTestId } = render(<Import navigation={navigationMock} />)
    fireEvent(getByTestId('google-takeout-link'), new NativeTestEvent('press'));
    expect(openURLSpy).toHaveBeenCalled();
  });

  it('clicking on import Takeout button opens a file picker', async () => {
    const {getByTestId} = render(<Import navigation={navigationMock} />)
    await act(async () => {
      await fireEvent(getByTestId('google-takeout-import-btn'), new NativeTestEvent('press'));
    });
    expect(pickFileSpy).toHaveBeenCalled();
    expect(importTakeoutDataSpy).toHaveBeenCalled();
  });
})

