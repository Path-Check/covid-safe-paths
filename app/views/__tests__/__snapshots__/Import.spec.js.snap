// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import component renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1f2c9b",
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1f2c9b",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <SvgXml
          color="#ffffff"
          style={
            Object {
              "paddingTop": 12,
            }
          }
          xml="
  <svg width=\\"23\\" height=\\"24\\" viewBox=\\"0 0 23 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M11.4854 3L2.22803 12.2573L11.4854 21.5146\\" stroke=\\"currentColor\\" stroke-opacity=\\"0.4\\" stroke-width=\\"3\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" />
  </svg>
"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2e2e2e",
              "fontFamily": "IBMPlexSans",
              "fontSize": 17,
              "lineHeight": 24,
            },
            Object {
              "writingDirection": "ltr",
            },
            Object {
              "color": "#ffffff",
              "fontFamily": "IBMPlexSans-Bold",
              "fontSize": 19,
              "fontWeight": "500",
              "lineHeight": 32,
              "textTransform": "uppercase",
            },
          ]
        }
      >
        Import Locations
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#f8f8ff",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 17,
                "lineHeight": 24,
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "marginBottom": 16,
              },
            ]
          }
        >
          Adding location data from Google will give you a head start on building your recent locations.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 17,
                "lineHeight": 24,
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "marginBottom": 16,
              },
            ]
          }
        >
          Before you can import, you must first "Take out" your location data from Google.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2e2e2e",
                "fontFamily": "IBMPlexSans",
                "fontSize": 17,
                "lineHeight": 24,
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "marginBottom": 16,
              },
            ]
          }
        >
          Visit Google Takeout and export your Location History using the following settings: 
1. Delivery method: "Add to Drive" 
2. Frequency: "Export once" 
3. File type & size: ".zip" and "1GB"
4. Google sends an email when the export is ready 
5. Return here to import locations from Google Drive
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#4051db",
              "borderColor": "#4051db",
              "borderRadius": 8,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 20,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 16,
            }
          }
          testID="google-takeout-link"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2e2e2e",
                  "fontFamily": "IBMPlexSans",
                  "fontSize": 17,
                  "lineHeight": 24,
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#ffffff",
                  "fontSize": 19,
                  "fontWeight": "500",
                },
              ]
            }
          >
            Visit Google Takeout
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#4051db",
              "borderColor": "#4051db",
              "borderRadius": 8,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 20,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 16,
            }
          }
          testID="google-takeout-import-btn"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2e2e2e",
                  "fontFamily": "IBMPlexSans",
                  "fontSize": 17,
                  "lineHeight": 24,
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#ffffff",
                  "fontSize": 19,
                  "fontWeight": "500",
                },
              ]
            }
          >
            Import Locations
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
